include/tvm/relay/pass.h

```diff
@@ -390,9 +368,36 @@ TVM_DLL Expr ToGraphNormalForm(const Expr& e);
  * It will do as much computation in compile time as possible.
  * It has two benefit: remove runtime overhead, and allow more optimization (typically fusion).
  * As a side effect, code size will explode.
+ *
+ * \param e the expression,
+ *
+ * \return the optimized expression.
  */
 Expr PartialEval(const Expr& e);
 
+/*! \brief A hashing structure in the style of std::hash. */
+struct StructuralHash {
+  /*! \brief Hash a Relay type.
+   *
+   * Implements structural hashing of a Relay type.
+   *
+   *  \param type the type to hash.
```

Comment:
seems the location of StructualHash does not have to change

Reply:
I wanna move the big one down, and have the small one together, so it is easier to read.